server/src/test/java/org/elasticsearch/ingest/IngestServiceTests.java

```diff
@@ -812,6 +812,27 @@ public void testUpdatingStatsWhenRemovingPipelineWorks() {
         assertThat(ingestService.stats().getStatsPerPipeline(), not(hasKey("_id2")));
     }
 
+    public void testExecuteWithDrop() {
+        IngestService ingestService = createWithProcessors(Collections.singletonMap("drop", new DropProcessor.Factory()));
+        PutPipelineRequest putRequest = new PutPipelineRequest("_id",
+            new BytesArray("{\"processors\": [{\"drop\" : {}}]}"), XContentType.JSON);
```

Comment:
minor: you may also want a second (mock) processor here, and assert that the second processor never executes.